runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -739,17 +742,32 @@ function M.get_virtual_text_chunks_for_line(bufnr, line, line_diags, opts)
   end
   local last = line_diags[#line_diags]
 
-  -- TODO(tjdevries): Allow different servers to be shown first somehow?
-  -- TODO(tjdevries): Display server name associated with these?
   if last.message then
+    local server_name = ''
+
+    if show_server_name then
+      local active_clients = vim.lsp.get_active_clients()
+      if not vim.tbl_isempty(active_clients) then
+        for _,client in ipairs(active_clients) do
+          if diagnostic_cache_lines[bufnr][client.id][line] ~= nil then
+            server_name = client.config.cmd[1]:match('[^\\/]+$')
+          end
+        end
+      end
```

Comment:
I think it would be easier to just pass the `client_id` as an (optional) argument

Reply:
hmmm the function will receive five arguments ,It's too long?

Reply:
Actually, `line_diags` already have a `source` prop directly from LSP that you can use.
https://microsoft.github.io/language-server-protocol/specifications/specification-current/#diagnostic

Maybe default to the LSP name only when that is null?

Reply:
source just describe string like  in
 lua-language-server  is `source = "Lua Syntax Check." `, gopls is  `source = "compiler"`, I don't think add this to last is useful...

Reply:
ah yeah that's not that useful.. I only tested with pyright, which sets them to strings like `Pyright (reportGeneralTypeIssues)`